pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -111,22 +126,49 @@ protected void setUpTask() {
    */
   @Override
   public final void run() {
-    _running = true;
-
-    if (_started) {
-      long startTime = System.currentTimeMillis();
-      LOGGER.info("Start running task: {}", _taskName);
-      try {
-        runTask();
-      } catch (Exception e) {
-        LOGGER.error("Caught exception while running task: {}", _taskName, e);
+    try {
+      // Don't allow a task to run more than once at a time.
+      _runLock.lock();
+      _running = true;
+
+      String periodicTaskRequestId =
+          _activePeriodicTaskProperties != null ? _activePeriodicTaskProperties.getProperty("requestid") : null;
```

Comment:
import from a static string define instead of hard-coding "requestid"

Reply:
Fixed.